### ISSUE TITLE
firedrake does not work with external petsc

### ISSUE THREAD

**Issue by tkarna** [firedrakeproject/firedrake#1571]
I have installed firedrake on our supercomputer by first building petsc and using `--honour-petsc-dir`. Firedrake install finishes successfully but when I import firedrake I get an error:
```
Traceback (most recent call last):
  File "stommel2d.py", line 13, in <module>
    from thetis import *
  File "/fmi/scratch/project_2001635/karnatuo/sources/firedrake/firedrake/src/thetis/thetis/__init__.py", line 2, in <module>
    from thetis.utility import *
  File "/fmi/scratch/project_2001635/karnatuo/sources/firedrake/firedrake/src/thetis/thetis/utility.py", line 5, in <module>
    from .firedrake import *
  File "/fmi/scratch/project_2001635/karnatuo/sources/firedrake/firedrake/src/thetis/thetis/firedrake.py", line 6, in <module>
    from firedrake import *  # NOQA
  File "/fmi/scratch/project_2001635/karnatuo/sources/firedrake/firedrake/src/firedrake/firedrake/__init__.py", line 51, in <module>
    from firedrake.assemble import *
  File "/fmi/scratch/project_2001635/karnatuo/sources/firedrake/firedrake/src/firedrake/firedrake/assemble.py", line 17, in <module>
    from firedrake.slate import slac
srun: Job step aborted: Waiting up to 32 seconds for job step to finish.
  File "/fmi/scratch/project_2001635/karnatuo/sources/firedrake/firedrake/src/firedrake/firedrake/slate/slac/__init__.py", line 1, in <module>
    from firedrake.slate.slac.compiler import *  # noqa: F401
  File "/fmi/scratch/project_2001635/karnatuo/sources/firedrake/firedrake/src/firedrake/firedrake/slate/slac/compiler.py", line 56, in <module>
    with open(filepath) as file:
FileNotFoundError: [Errno 2] No such file or directory: '/fmi/scratch/project_2001635/karnatuo/sources/firedrake/firedrake/src/petsc/default/lib/petsc/conf/petscvariables'
```
I.e. slate is erronously looking for petsc in the default location. My workaround was to link petsc install dir to that location.

**Comment by tkarna** [comment]
Also, `firedrake-install` now links `mpicc` etc compilers to `firedrake/bin` directory. This does not work with Intel MPI. If one uses gcc with Intel MPI, `mpicc` actually just calls `mpigcc` which is assumed to reside in the same directory:
```
/fmi/scratch/project_2001635/karnatuo/sources/firedrake/firedrake/bin/mpicc: line 54: /fmi/scratch/project_2001635/karnatuo/sources/firedrake/firedrake/bin/mpigcc: No such file or directory
```
My workaround was to remove the `mpi*` links from `firedrake/bin`.

**Comment by tkarna** [comment]
Lastly, firedrake-install `--show-petsc-configure-options` seems to be broken:
```
Running firedrake-install --show-petsc-configure-options
*********************************************
Would build PETSc with the following options:
*********************************************

Traceback (most recent call last):
  File "firedrake-install", line 1077, in <module>
    log.info("\n".join(os.environ["PETSC_CONFIGURE_OPTIONS"].split()))
  File "/usr/lib/python3.6/os.py", line 669, in __getitem__
    raise KeyError(key) from None
KeyError: 'PETSC_CONFIGURE_OPTIONS'
```

**Comment by wence-** [comment]
> I.e. slate is erronously looking for petsc in the default location. My workaround was to link petsc install dir to that location.

This is fixed by #1568 

The second part, I think that we should not set up the MPI environment if honour-petsc-dir is passed and the mpi compiler names are not specified.

The third part, that looks like an bug in ordering of things occuring in firedrake-install.

**Comment by tkarna** [comment]
Thanks, I confirm that `--honour-petsc-dir` works now.
